sycl/source/detail/pi_opencl.cpp

```diff
@@ -59,13 +59,16 @@ pi_result OCL(piDevicesGet)(pi_platform      platform,
   return cast<pi_result>(result);
 }
 
-pi_result OCL(piextDeviceSelectBinary)(
-  pi_device           device, // TODO: does this need to be context?
-  pi_device_binary *  images,
-  pi_uint32           num_images,
-  pi_device_binary *  selected_image) {
-
-  // TODO dummy implementation.
+pi_result OCL(piextDeviceSelectBinary)(pi_device device,
+                                       pi_device_binary *images,
+                                       pi_uint32 num_images,
+                                       pi_device_binary *selected_image) {
+
+  // TODO: this is a bare-bones implementation for choosing a device image
```

Comment:
I wonder how this implementation is tested.

Reply:
That's spot-on. Since the functionality is extended, the coverage should be updated accordingly, especially regression-wise. I'll update later.

Reply:
Extensive testing added.